checkrebuild

```diff
@@ -22,17 +22,24 @@ get_unofficial_pkgs() {
     comm -23 <(pacman -Qq | sort) <(pacman -Sql "${repos_to_skip[@]}" 2>/dev/null | sort)
 }
 
-filter_executable() {
-    while read -r file; do
-        [[ -f "$file" && -x "$file" ]] && echo "$file"
-    done
+newline2null(){
+    perl -pe 's/\n/\0/'
 }
 
 get_package_files() {
     xargs pacman -Qql |
-    perl -pe 's/\n/\0/' |
-    xargs -0 readlink -f |
-    filter_executable
+    newline2null |
+    xargs -0 readlink -ez |
+    sort -uz
+}
+
+filter_executable() {
+    LANG=C xargs -0 stat --printf "%F %a\t%n\n" |
+    grep -P "^regular file ([1-6]|)[1357]" |
+    cut -f2- |
+    newline2null |
```

Comment:
Once we switch to null-byte-separated string we really should avoid switching back to newline-separated strings at any point, or this will break if file paths contain newline characters...

I believe something like this should work:

```
LANG=C xargs -0 stat --printf "%F %a\t%n\0" |  # ends with \0
grep -Pzo "^regular file [1357].*\t\K.*" |     # regex cuts off down to file name only
```

Reply:
Slick move.